peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
+Neither of these examples is possible with today's f-strings. By providing a 
+mechanism to intercept and transform interpolated values, template strings
+enable a wide range of string processing use cases.
 
-The f-string interpolation of ``{name}`` leads to the new machinery in tag
-strings:
 
-- ``args[0]`` is still the string-like ``'Hello '``, this time with a trailing space
-- ``args[1]`` is an expression -- the ``{name}`` part
-- Tag strings represent this part as an *interpolation* object as discussed below
+Specification
+=============
+
+Template String Literals
+------------------------
 
-The ``*args`` list is a sequence of ``Decoded`` and ``Interpolation`` values. A "decoded" object
-is a string-like object with extra powers, as described below. An "interpolation" object is a
-tuple-like value representing how Python processed the interpolation into a form useful for your
-tag function. Both are fully described below in `Specification`_.
+This PEP introduces a new string prefix, ``t``, to define template string literals.
+These literals resolve to a new type, ``types.Template``.
 
-Here is a more generalized version using structural pattern matching and type hints:
+The following code creates a ``Template`` instance:
 
 .. code-block:: python
 
-    from typing import Decoded, Interpolation  # Get the new protocols
+    from types import Template
+    template = t"This is a template string."
+    assert isinstance(template, Template)
 
-    def greet(*args: Decoded | Interpolation) -> str:
-        """Handle arbitrary args using structural pattern matching."""
-        result = []
-        for arg in args:
-            match arg:
-                case Decoded() as decoded:
-                    result.append(decoded)
-                case Interpolation() as interpolation:
-                    value = interpolation.getvalue()
-                    result.append(value.upper())
+Template string literals support the full syntax of :pep:`701`. This includes
+the ability to nest template strings within interpolations, as well as the ability
+to use any quote mark in the string literal. Like other string prefixes, the template
+prefix must immediately precede the quote mark. Like f-strings, both lowercase 
+``t`` and uppercase ``T`` prefixes are supported. Like f-strings, t-strings may
+not be combined with the ``b`` or ``u`` prefixes. They *may* be combined with 
+the ``r`` prefix; see the `Raw Template Strings`_ section below for more information.
 
-        return f"{''.join(result)}!"
 
-    name = "World"
-    greeting = greet"Hello {name} nice to meet you"
-    assert greeting == "Hello WORLD nice to meet you!"
+The ``Template`` Type
+---------------------
 
-Tag strings extract more than just a callable from the ``Interpolation``. They also
-provide Python string formatting info, as well as the original text:
+Template strings evaluate to an instance of a new type, ``Template``, found
+in the ``<TBC>`` module (proposed: ``types``):
```

Comment:
You've used `types.Template` elsewhere
```suggestion
Template strings evaluate to an instance of a new type, ``types.Template``:
```

Reply:
I'll fix this for now, but I have a feeling we'll want to revisit/bikeshed this a bit.

@lysnikolaou and I aren't in love with `types.Template` but I don't yet have a better suggestion.